docs/spec_draft.md

```diff
@@ -358,6 +359,56 @@ IEEE-754 specification.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.concatenate
+
+### Semantics
+
+Concatenates sequence of tensors in `val` along dimension `dimension` and
+produces a tensor as a result. The tensors are of the same rank as the input
+operands except the specified dimension.
+
+### Operands
+
+| Name        | Type                                                                     |
+|-------------|--------------------------------------------------------------------------|
+| `dimension` | `i64`                                                                    |
+| `val`       | Sequence of tensors of integer, floating-point, or complex element types |
+
+### Results
+
+| Name     | Type                                                        |
+|----------|-------------------------------------------------------------|
+| `result` | tensor of integer, floating-point, or complex element types |
+
+### Constraints
+
+  * (C1) `val` and `result` have the same type.
+  * (C2) All tensors in `val` have the same element type and rank.
+  * (C3) Tensors of `val` all have the same dimension except for `dimension`.
+
+### Examples
+
+```mlir
+// 1-dimensional concatenate
+
+// %val1: [1, 2]
+// %val2: [3, 4]
+// %val3: [5, 6]
+// %dimension: 0
+%result = "stablehlo.concatenate"(%val1, %val2, %val3, %dimension) : (tensor<2xi32>, tensor<2xi32>, tensor<2xi32>, i64) -> tensor<8xi32>
```

Comment:
The actual MLIR syntax is different. `dimension` will be passed as an attribute, along the lines of `{ dimension = 0 : i64 }`. The type signature will be different as well.